src/extensions/templateExtension.ts

```diff
@@ -0,0 +1,91 @@
+import { TemplateContext, ScopeData } from "../compilation";
```

Comment:
Please remove this file.
There's a lot of logic here that assumes a certain type of extensions. I wouldn't want to limit other potential extensions. Also, see my comment on ITemplatePlugin.

Reply:
Agreed